scripts/demangle.py

```diff
@@ -20,15 +20,41 @@ def get_source_from_docker(
     container = client.containers.run(image, detach=True)
     # check file_path, it may not the absolute path
     if "/" not in file_path or file_path[0] != "/":
+        file_path4 = "/src"
+        file_path3 = os.path.join("/src", file_path)
+        file_path2 = os.path.join("/src", proj_name, "src", file_path)
         file_path = os.path.join("/src", proj_name, file_path)
-    # another possible path is os.path.join("/src", proj_name, "src", file_path)
+
+        build_sh = ""
+        for chunk in container.get_archive("/src/build.sh")[0]:
+            build_sh += chunk.decode("utf-8")
+        pattern = r"cd\s+(\S+)"
+        matches = re.findall(pattern, build_sh)
+        # Print the extracted strings
+        for match in matches:
+            file_path4 = os.path.join(file_path4, match)
     try:
         code_content = ""
         for chunk in container.get_archive(file_path)[0]:
             code_content += chunk.decode("utf-8")
     except docker.errors.NotFound:
```

Comment:
Nesting `try-except` is not a good idea, the code inside is repeated as well. Is there an API that does something similar to `container.exists(path)`? If so, please use `if-else` branch to confirm existence first and `get_archive` later.

Reply:
You are right. However, I have tried several methods to check whether the path exits before, but they are not succeed. One way I was doing this is using `container.exec_run(f"test -e {path} && echo 1 || echo 0")` But the docker raise error (docker.errors.APIError: 500 Server Error Internal Server Error ("Container is not running: Exited (1) Less than a second ago"))